continuous-pipe.yml.erb

```diff
@@ -0,0 +1,192 @@
+                memory: 250Mi
+              limits:
+                cpu: 500m
+                memory: 2Gi
+
+            environment_variables:
+              MYSQL_PASSWORD: ${DATABASE_PASSWORD}
+              MYSQL_ROOT_PASSWORD: ${DATABASE_ROOT_PASSWORD}
+              MYSQL_DATABASE: ${DATABASE_NAME}
+              MYSQL_USER: ${DATABASE_USER}
+
+  reserve_ip:
+    deploy:
+      services:
+        web:
+          specification:
+            source:
+              image: continuouspipe/landing-page
+              tag: latest
+
+            accessibility:
+              from_external: true
+
+            ports:
+              - 80
+              - 443
+
+      filter:
+        expression: 'tasks.storage.services.database.created'
+
+  setup:
+    run:
+      image:
+        from_service: web
+
+      commands:
+        - container setup
+
+      environment_variables: &WEB_ENV_VARS
+        APP_USER_LOCAL: false
+        DATABASE_NAME: ${DATABASE_NAME}
+        DATABASE_USER: ${DATABASE_USER}
+        DATABASE_PASSWORD: ${DATABASE_PASSWORD}
+        DATABASE_ROOT_PASSWORD: ${DATABASE_ROOT_PASSWORD}
+        AUTH_HTTP_ENABLED: true
+        AUTH_HTTP_HTPASSWD: ${AUTH_HTTP_HTPASSWD}
+        AUTH_IP_WHITELIST_ENABLED: true
+        AUTH_HTTP_WHITELIST_IPS: ${AUTH_HTTP_WHITELIST_IPS}
+
+    filter:
+      expression: 'tasks.storage.services.database.created and code_reference.branch not in ["production", "uat"]'
+
+  deployment:
+    deploy:
+      services:
+        web:
+          specification:
+            accessibility:
+              from_external: true
+
+            environment_variables:
+              <<: *WEB_ENV_VARS
+
+            resources:
+              requests:
+                cpu: 50m
+                memory: 50Mi
+
+              limits:
+                cpu: 1
+                memory: 2G
+
+            ports:
+              - 80
+              - 443
+
+          deployment_strategy:
+            readiness_probe:
+              type: tcp
+              port: 80
+
+            liveness_probe:
+              type: exec
+              command:
+                - curl
+                - -k
+                - -I
+                - -X GET
+                - -H
+                - "Authorization: Basic ${AUTH_HTTP_AUTHORISATION_STRING}"
+                - https://localhost/
+              initial_delay_seconds: 30
+              period_seconds: 300 # 5 minutes
+              success_threshold: 1
+              failure_threshold: 5
+
+        cron:
+          specification:
+            accessibility:
+              from_cluster: false
+
+            source:
+              from_service: web
+
+            ports: []
+
+            environment_variables:
+              <<: *WEB_ENV_VARS
+              START_MODE: cron
+
+            resources:
+              requests:
+                cpu: 50m
+                memory: 20Mi
+
+              limits:
+                cpu: 1
+                memory: 1G
+
+pipelines:
```

Comment:
Could we move this to the top above `tasks:` so it makes for a more readable file?